test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/ODataUriParserTests.cs

```diff
@@ -1183,22 +1183,6 @@ public void ParseComposableEscapeFunctionUrlAndPropertyReturnsTheSameODataPath()
             Assert.Equal("Name", proSegment.Property.Name);
         }
 
-        [Fact]
-        public void ParseEscapeFunctionUrlThrowsWithoutEscapeFunction()
-        {
-            // Arrange
-            IEdmModel model = GetEdmModelWithEscapeFunction(escape: false);
-
-            // Act
-            var fullUriString = ServiceRoot + "/root:/photos/2018/February";
-            var parser = new ODataUriParser(model, ServiceRoot, new Uri(fullUriString));
-            Action test = () => parser.ParsePath();
-
-            // Assert
-            var odataException = Assert.Throws<ODataException>(test);
-            Assert.Equal(ODataErrorStrings.RequestUriProcessor_NoBoundEscapeFunctionSupported("NS.OneDrive"), odataException.Message);
-        }
-

```

Comment:
What is the behavior for this url? It should still throw an invalid url, right? because root: is not a valid identifier?

Reply:
Correct. Added the test back. 

---
In reply to: [366144030](https://github.com/OData/odata.net/pull/1621#discussion_r366144030) [](ancestors = 366144030)